.circleci/config.yml

```diff
@@ -2,213 +2,193 @@ version: 2
-          git clone --branch v0.4.0 https://github.com/googleapis/gnostic $GOPATH/src/k8s.io/client-go/vendor/github.com/googleapis/gnostic
-
-    # CircleCI recommends caches be ~500MB. We split the cache into
-    # two parts to achieve this.
-    # We specifically do not cache gopkg.in to avoid caching Datadog
-    # We can cache other packages here or wait to move to go modules
-    # in the future.
-    - restore_cache:
-        keys:
-          - go-pkg-part-1-cache-v1-{{ checksum "ddtrace/Gopkg.toml" }}
-          - go-pkg-part-1-cache-v1
-    - restore_cache:
-        keys:
-          - go-pkg-part-2-cache-v2-{{ checksum "ddtrace/Gopkg.toml" }}
-          - go-pkg-part-2-cache-v2
-    - run:
-        name: Fetching dependencies
+        name: build
         command: |
-          go get -v -t ./...
-          go get -v -u golang.org/x/lint/golint
-          go get -v -u github.com/alecthomas/gometalinter
-    - save_cache:
-        key: go-pkg-part-1-cache-v1-{{ checksum "ddtrace/Gopkg.toml" }}
-        paths:
-          - /go/src/github.com
-          - /go/src/golang.org
-    - save_cache:
-        key: go-pkg-part-2-cache-v2-{{ checksum "ddtrace/Gopkg.toml" }}
-        paths:
-          - /go/src/cloud.google.com
-          - /go/src/google.google.org
-          - /go/src/go.mongodb.org
-          - /go/src/go.opencensus.io
-          - /go/src/golang.org
-          - /go/src/sigs.k8s.io
-          - /go/src/k8s.io
-
-    - run:
-        name: Vendor klog v0.4.0
-        # Temporary, until kubernetes/client-go#656 gets resolved.
-        command: >
-          git clone --branch v0.4.0
-          https://github.com/kubernetes/klog
-          $GOPATH/src/k8s.io/client-go/vendor/k8s.io/klog
+          go build ./ddtrace/...
 
-    - run:
-        name: Wait for MySQL
-        command: dockerize -wait tcp://localhost:3306 -timeout 1m
-
-    - run:
-        name: Wait for Postgres
-        command: dockerize -wait tcp://localhost:5432 -timeout 1m
-
-    - run:
-        name: Wait for Redis
-        command: dockerize -wait tcp://localhost:6379 -timeout 1m
-
-    - run:
-        name: Wait for ElasticSearch (1)
-        command: dockerize -wait http://localhost:9200 -timeout 1m
-
-    - run:
-        name: Wait for ElasticSearch (2)
-        command: dockerize -wait http://localhost:9201 -timeout 1m
-
-    - run:
-        name: Wait for Datadog Agent
-        command: dockerize -wait tcp://127.0.0.1:8126 -timeout 1m
+    - persist_to_workspace:
+        root: /home/circleci/go
+        paths:
+          - src
+          - pkg
+          - bin
 
-    - run:
-        name: Wait for Cassandra
-        command: dockerize -wait tcp://localhost:9042 -timeout 2m
+  test-core:
+    working_directory: /home/circleci/go/src/gopkg.in/DataDog/dd-trace-go.v1
+    docker:
+      - image: circleci/golang:1.13
+        environment:
+          GO111MODULE: "on"
+          GOPATH: "/home/circleci/go"
```

Comment:
lets move all `docker` to one executor and use there.

we have building on go1.12 and testing on go1.13. Might be better to have them all on the same version

Reply:
oh nvm i saw the conversation re different golang versions. 

Still think listing the executors on top might be cool: https://circleci.com/docs/2.0/configuration-reference/#executors-requires-version-21

Reply:
Some of the contrib packages require go1.13 (`k8s.io/client-go`), but we would prefer to test at least the tracer (`ddtrace/tracer`) with the minimum supported version - go1.12

Reply:
@knusbaum makes sense.

Reply:
Sorry @roopakv, somehow your second message didn't show up when I was responding. I only saw the first comment.

Reply:
Might be something to look into if it saves CI time and wasted resources. It's easy to get go1.12 using `go get golang.org/dl/go1.12` just for the build. The base executor image could have both.